framework/src/timesteppers/CompositionDT.C

```diff
@@ -0,0 +1,178 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CompositionDT.h"
+#include "MooseApp.h"
+#include "Transient.h"
+#include "TimeSequenceStepperBase.h"
+
+registerMooseObject("MooseApp", CompositionDT);
+
+InputParameters
+CompositionDT::validParams()
+{
+  InputParameters params = TimeStepper::validParams();
+
+  params.addRequiredParam<std::string>(
+      "base_timestepper", "Provide a time stepper type to produce a base time step size");
+
+  params.addParam<std::vector<std::string>>(
+      "input_timesteppers",
+      "The input TimeSteppers to compose time step size.  This can either be N timesteppers or 1 "
```

Comment:
base + input wont work for "average"

maybe:
"average_step_from ="
"maximum_step_from ="
"minimum_step_from ="

Reply:
This part is done. I remove `input_timesteppers` and use `maximum_step_from`/`maximum_step_from` instead.